Develop

Senkid12 opened the pull request:
Выполнил задание 7-ого спринта, жду ревью), Буду рад рекомендациям по улучшению кода!)

senkid committed:
add allure report

senkid committed:
first-commit for pull-request

senkid committed:
second-commit for pull-request

senkid committed:
second-commit for pull-request

alesyaisachenko commented on src/main/java/org/yandex/praktikum/model/courier/courierCreate/Courier.java:
```diff
@@ -0,0 +1,41 @@
+package org.yandex.praktikum.model.courier.courierCreate;
+
+public class Courier {
```
Можно лучше:

можно использовать @Data из библиотеки Lombok

alesyaisachenko commented on src/main/java/org/yandex/praktikum/model/courier/courierCreate/CourierWithoutLogin.java:
```diff
@@ -0,0 +1,30 @@
+package org.yandex.praktikum.model.courier.courierCreate;
+
+public class CourierWithoutLogin {
```
Это нужно доработать:

вместо этого класса нужно использовать класс Courier.
Класс CourierWithoutLogin нужно удалить.

alesyaisachenko commented on src/main/java/org/yandex/praktikum/model/courier/courierCreate/CourierWithoutName.java:
```diff
@@ -0,0 +1,30 @@
+package org.yandex.praktikum.model.courier.courierCreate;
+
+public class CourierWithoutName {
```
Это нужно доработать:

вместо этого класса нужно использовать класс Courier.
Класс CourierWithoutName нужно удалить.

alesyaisachenko commented on src/main/java/org/yandex/praktikum/model/courier/courierCreate/CourierWithoutPassword.java:
```diff
@@ -0,0 +1,30 @@
+package org.yandex.praktikum.model.courier.courierCreate;
+
+public class CourierWithoutPassword {
```
Это нужно доработать:

вместо этого класса нужно использовать класс Courier.
Класс CourierWithoutPassword нужно удалить.

alesyaisachenko commented on src/main/java/org/yandex/praktikum/model/courier/courierForAuthorization/CourierForAuthorization.java:
```diff
@@ -0,0 +1,30 @@
+package org.yandex.praktikum.model.courier.courierForAuthorization;
+
+public class CourierForAuthorization {
```
Это нужно доработать:

вместо этого класса нужно использовать класс Courier.
Класс CourierForAuthorization нужно удалить.

alesyaisachenko commented on src/main/java/org/yandex/praktikum/model/courier/courierForAuthorization/CourierForAuthorizationWithoutLogin.java:
```diff
@@ -0,0 +1,26 @@
+
+import org.yandex.praktikum.model.courier.courierCreate.Courier;
+
+public class CourierForAuthorizationWithoutLogin {
```
Это нужно доработать:

вместо этого класса нужно использовать класс Courier.
Этот класс нужно удалить.

alesyaisachenko commented on src/main/java/org/yandex/praktikum/model/courier/courierForAuthorization/CourierForAuthorizationWithoutPassword.java:
```diff
@@ -0,0 +1,25 @@
+
+import org.yandex.praktikum.model.courier.courierCreate.Courier;
+
+public class CourierForAuthorizationWithoutPassword {
```
Это нужно доработать:

вместо этого класса нужно использовать класс Courier.
Этот класс нужно удалить.

alesyaisachenko commented on src/test/java/org/yandex/praktikum/CourierCreateTest.java:
```diff
@@ -0,0 +1,139 @@
+    }
+
+    @Step("Создание курьера")
+    public Response create(Courier courier) {
```
Это нужно обязательно исправить: 

нужно улучшить архитектуру кода.

Сейчас в каждом тесте ты дублируешь код по формированию и отправления запроса.

Дублирование кода нужно избегать. Как это сделать:
Создать классы CourierApi и OrderApi.
В каждом классе создать методы для формирования запроса (например: создание courier, удаление courier, логин и т.д.).  Проверку ответа делать или в тесте, или можно написать отдельные методы.
В этом случае, если изменится запрос для создания курьера, тебе придется вносить правки в одном месте (в методе), а не во все тесты сразу.

alesyaisachenko replied:
Этот момент еще нужно доработать.

alesyaisachenko commented on src/test/java/org/yandex/praktikum/CourierCreateTest.java:
```diff
@@ -0,0 +1,139 @@
+    }
+
+    @Step("Создание курьера без поля password")
+    public Response createWithoutPassword(CourierWithoutPassword courierWithoutPassword) {
```
Это нужно доработать:

этот метод нужно удалить, использовать вместо него метод --create(Courier courier)

alesyaisachenko commented on src/test/java/org/yandex/praktikum/CourierCreateTest.java:
```diff
@@ -0,0 +1,139 @@
+    }
+
+    @Step("Создание курьера без поля login")
+    public Response createWithoutLogin(CourierWithoutLogin courierWithoutLogin) {
```
Это нужно доработать:

этот метод нужно удалить, использовать вместо него метод --create(Courier courier)

alesyaisachenko commented on src/test/java/org/yandex/praktikum/CourierCreateTest.java:
```diff
@@ -0,0 +1,139 @@
+    }
+
+    @Step("Создание курьера без поля firstName")
+    public Response createWithoutName (CourierWithoutName courierWithoutName) {
```
Это нужно доработать:

этот метод нужно удалить, использовать вместо него метод --create(Courier courier)

alesyaisachenko commented on src/test/java/org/yandex/praktikum/CourierCreateTest.java:
```diff
@@ -0,0 +1,139 @@
+                        .post("/api/v1/courier");
+        return response;
+    }
+//    @Step("Удаление курьера")
```
Это нужно доработать:

отладочный код нужно удалять

alesyaisachenko replied:
Это тоже нужно доработать

alesyaisachenko commented on src/test/java/org/yandex/praktikum/CourierForAuthorizationTest.java:
```diff
@@ -0,0 +1,124 @@
+
+    }
+    // Подумал может есть какой-то способ как-то правильней выводить информацию с теста
+    // (думаю что sout не используются в тестах), буду благодарен за подсказку как это грамотней сделать)
```
Привет!

у rest-assured есть метод .log(), можно логировать весь запрос и \ или ответ (весь, только хедеры, только боди  и т.д.)

alesyaisachenko commented on src/test/java/org/yandex/praktikum/CourierForAuthorizationTest.java:
```diff
@@ -0,0 +1,124 @@
+        CourierForAuthorizationWithoutPassword courierForAuthorizationWithoutPassword = courierForAuthorizationGenerator.getCourierForAuthorizationWithoutPassword(courierGenerator.getCourier());
+        Response response = authorizationCourierWithoutPassword(courierForAuthorizationWithoutPassword);
+        System.out.println("Ответ с сервера " + response.body().asString());
+        response.then().assertThat().body("message", equalTo("Недостаточно данных для входа"));
```
Это нужно доработать:

сейчас тест падает на этом шаге -- это нужно пофиксить

alesyaisachenko commented on src/test/java/org/yandex/praktikum/CourierForAuthorizationTest.java:
```diff
@@ -0,0 +1,124 @@
+        CourierForAuthorization courierForAuthorizationWithPasswordNull = courierForAuthorizationGenerator.getCourierForAuthorizationWithPasswordNull(courierGenerator.getCourier());
+        Response response = authorizationCourier(courierForAuthorizationWithPasswordNull);
+        System.out.println("Ответ с сервера " + response.body().asString());
+        response.then().assertThat().body("message", equalTo("Недостаточно данных для входа"));
```
Это нужно доработать:

сейчас тест падает на этом шаге -- это нужно пофиксить

alesyaisachenko commented on src/test/java/org/yandex/praktikum/CourierCreateTest.java:
```diff
@@ -0,0 +1,139 @@
+    public void createCourier() {
+        Response response = create(courierGenerator.getCourier());
+        System.out.println("Ответ с сервера " + response.body().asString());
+        response.then().assertThat().body("ok", equalTo(true));
```
Это нужно доработать во всем коде:

в тесте нужно проверять и код ответа и боди.

Подправь, плиз, во всех тестах

alesyaisachenko commented on src/test/java/org/yandex/praktikum/CourierCreateTest.java:
```diff
@@ -0,0 +1,139 @@
+                        .header("Content-type", "application/json")
+                        .body(courier)
+                        .when()
+                        .post("/api/v1/courier");
```
Можно лучше :

"/api/v1/courier" -- лучше было вынести в final static переменную.

alesyaisachenko commented on src/test/java/org/yandex/praktikum/OrdersGetTest.java:
```diff
@@ -0,0 +1,35 @@
+    public void getOrderByTrack() {
+        Response response = getOrders();
+        System.out.println("Ответ с сервера " + response.body().asString());
+        response.then().statusCode(200).and().body("orders", notNullValue());
```
Можно лучше:

статус коды можно было взять из библиотеки:
import  static  org.apache.http.HttpStatus.*;

statusCode(SC_CREATED)

senkid committed:
third-commit
1) использовал в классе Courier.java библиотеку lombok
2) удалил класс CourierWithoutLogin, CourierWithoutName, CourierWithoutPassword, CourierForAuthorization
3) Во всех тестах проверил код ответа и боди
4) Вынес ручки в класс Enpoints.java
5) Статус коды взял из библиотеки org.apache.http.HttpStatus.

senkid committed:
third-commit
1) использовал в классе Courier.java библиотеку lombok
2) удалил класс CourierWithoutLogin, CourierWithoutName, CourierWithoutPassword, CourierForAuthorization
3) Во всех тестах проверил код ответа и боди
4) Вынес ручки в класс Enpoints.java
5) Статус коды взял из библиотеки org.apache.http.HttpStatus.

alesyaisachenko commented on src/main/java/org/yandex/praktikum/model/order/Order.java:
```diff
@@ -0,0 +1,100 @@
+package org.yandex.praktikum.model.order;
+
+public class Order {
```
Можно лучше:
здесь по аналогии можно было использовать @Data
@AllArgsConstructor

alesyaisachenko commented on src/test/java/org/yandex/praktikum/CourierForAuthorizationTest.java:
```diff
@@ -0,0 +1,99 @@
+import static org.yandex.praktikum.constants.Endpoints.endpointForCourierAuthorization;
+import static org.yandex.praktikum.constants.Endpoints.endpointForCreateCourier;
+
+public class CourierForAuthorizationTest {
```
Это нужно доработать:

после прохождения тестов нужно удалять созданного тестового юзера

alesyaisachenko commented on src/main/java/org/yandex/praktikum/constants/Endpoints.java:
```diff
@@ -0,0 +1,7 @@
+package org.yandex.praktikum.constants;
+
+public class Endpoints {
+    public static final String endpointForCreateCourier = "/api/v1/courier";
```
Это нужно доработать:
1. энд-поинты есть, но они не используются в коде.
2. endpointForCreateCourier -- нужно END_POINT_FOR_CREATE_COURIER

senkid committed:
Fourth-commit
1) использовал в классе Order.java библиотеку lombok
2) После тестов с авторизацией добавил удаление тестового курьера
3) Исправил ручки в Endpoints.java
